### PR TITLE
Fullscreen Image

### LINKED ISSUE #1354: [2.0] Image jumps to the left when closing preview
**Describe the bug**
When closing the preview window of an image post by pulling down, the image jumps to the left before completing the animation. 

https://github.com/user-attachments/assets/f1ffb664-c380-49b4-bf69-e27e0b52cd9e

**To Reproduce**
Open any image post by clicking the thumbnail and close it by swiping down. 

**Mlem Information:**
Mlem 2.0 (68)
iPhone 14 Pro
iOS 18.0

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #1354 

# Pull Request Information

This PR replaces the sheet-based image viewer with an overlay-based implementation. It can be swiped away either up or down, and interactive dismissal works on iPad now too.

I've also moved the audio button to the bottom to avoid conflict with the close button.

### CODE REVIEW

**Comment by Sjmarf** [comment]
Swipe to dismiss doesn't seem to be working for me on my physical device 🤔

**Review by Sjmarf — CHANGES_REQUESTED** [review]
The old `NavigationPage.imageViewer` code needs removing, assuming it isn't still being used anywhere?

**Comment by EricBAndrews** [comment]
> The old NavigationPage.imageViewer code needs removing, assuming it isn't still being used anywhere?

It is still used--the overlay needs to be a `NavigationLayerView` to gain access to the share menu and proxy failure popup

File: Mlem/App/Views/Pages/ImageViewer.swift [hunk 1]
```diff
@@ -9,25 +9,104 @@ import SwiftUI
...
     var body: some View {
-        ZoomableContainer {
-            DynamicMediaView(url: url, playImmediately: true)
+        ZoomableContainer(isZoomed: $isZoomed) {
+            DynamicMediaView(url: url, cornerRadius: 0, playImmediately: true)
+        }
+        .offset(y: offset)
+//        .onAppear {
+//            updateDragDistance(0)
+//        }
```
Comment (Sjmarf) [inline Mlem/App/Views/Pages/ImageViewer.swift]: Dead code?
```suggestion
```

**Review by Sjmarf — APPROVED** [review]
Ah, I see 👍 LGTM!

**Pull request merged**